### PR TITLE
LOK-2711: Review changes

### PR DESCRIPTION
**Pull request by Mohammad-Junid**
## Description
Need a BFF to return list of metrics names and labels. For now it will be a hard coded list later on we will get these from cortex or db e.g 

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2711

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Comment by Mohammad-Junid** [comment]
FE request 

`query GetMetricsLabel {
  metricsLabels {
 metricList {
      displayName
      metricName
      category
      expression
      labels
      unitType
    }
  }
}`

image

File: rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public class MetricLabel {
     private String displayName;
     private String metricName;
     private String category;
-    private List<MetricLabelKeyValue> labels;
+    private String expression;
+    private Map<String, Object> labels;
```
Comment (cgorantla) [inline rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java:35]: we can directly use String[] or List<String> in place of Object here.

Comment (Mohammad-Junid) [reply rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java:35]: We add two labels which  data type is different  like below . If we defined data type explicitly like above  then  **nodeId**   will be skipped . I just add for arbitrary purpose. 

   ```
   "labels": {
        "serviceTypes": [
          "SNMP",
          "WSMAN"
        ],
        "nodeId": ""
      }
```

Comment (Mohammad-Junid) [reply rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java]: Done

File: rest-server/src/main/resources/metrics-labels.json [hunk 2]
```diff
@@ -4,60 +4,111 @@
...
-      "labels": [
-        { "value": "HTTPMonitor", "name": "HTTP" },
-        { "value": "HTTPSMonitor", "name": "HTTPS" },
-        { "value": "SSHMonitor", "name": "SSH" },
-        { "value": "NTPMonitor", "name": "NTP" },
-        { "value": "SNMPMonitor", "name": "SNMP" },
-        { "value": "ICMPMonitor", "name": "ICMP" }
-      ],
+      "expressions": {
+        "HTTPS": "response_time_msec%1$s[4m]",
```
Comment (cgorantla) [inline rest-server/src/main/resources/metrics-labels.json:59]: we don't need range on response_time_msec, this is always an instant metric

File: rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java [hunk 3]
```diff
@@ -31,6 +32,7 @@ public class MetricLabel {
     private String displayName;
     private String metricName;
     private String category;
-    private List<MetricLabelKeyValue> labels;
+    private Map<String, String> expressions;
```
Comment (cgorantla) [inline rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java:35]: i would rather have this as Class Expression {
                                            String serviceType;
                                             String expression;
                                           }
and List<Expression>

Comment (Mohammad-Junid) [reply rest-server/src/main/java/org/opennms/horizon/server/model/MetricLabel.java]: Done

File: rest-server/src/main/resources/metrics-labels.json (hunk 2 above)
Comment (Mohammad-Junid) [reply rest-server/src/main/resources/metrics-labels.json:59]: Done

File: rest-server/src/main/resources/metrics-labels.json [hunk 4]
```diff
@@ -4,60 +4,111 @@
...
     },
     {
       "displayName": "Memory Utilization (%) 24h",
       "metricName": "memory_utilization",
       "category": "nodestatus",
-      "labels": [
-        { "value": "SNMP", "name": "SNMP" }
-      ],
-      "unitType": "bytes"
+      "expressions": {
```
Comment (cgorantla) [inline rest-server/src/main/resources/metrics-labels.json:100]: probably need to update expressions in the json , right ?

Comment (Mohammad-Junid) [reply rest-server/src/main/resources/metrics-labels.json:100]: The updated JSON file not pushed.  Just push the updated one

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**